### PR TITLE
row_cache: Make fill_buffer() preemptable when cursor leads with dummy rows

### PR DESCRIPTION
**Pull request by tgrabiec**
fill_buffer() will keep scanning until _lower_bound_changed is true,
even if preemption is signaled, so that the reader makes forward
progress.

Before the patch, we did not update _lower_bound on touching a dummy
entry. The read will not respect preemption until we hit a non-dummy
row. If there is a lot of dummy rows, that can cause reactor stalls.

Fix that by updating _lower_bound on dummy entries as well.

Refs #8153.

Tested with perf_row_cache_reads:

```
$ build/release/test/perf/perf_row_cache_reads -c1 -m200M
Rows in cache: 0
Populating with dummy rows
Rows in cache: 373929
Scanning
read: 183.658966 [ms], preemption: {count: 848, 99%: 0.545791 [ms], max: 0.519343 [ms]}, cache: 99/100 [MB]
read: 120.951515 [ms], preemption: {count: 257, 99%: 0.545791 [ms], max: 0.518795 [ms]}, cache: 99/100 [MB]
```

Notice that max preemption latency is low in the second "read:" line.

### CODE REVIEW

**Comment by tgrabiec** [comment]
In v2:
 - added missing include to perf_row_cache_reads, appeared after rebase

File: cache_flat_mutation_reader.hh [hunk 1]
```diff
@@ -620,6 +620,11 @@ void cache_flat_mutation_reader::add_to_buffer(const partition_snapshot_row_curs
         _read_context->cache().on_row_hit();
         add_clustering_row_to_buffer(mutation_fragment(*_schema, _permit, row.row(_read_context->digest_requested())));
     } else {
+        position_in_partition::less_compare less(*_schema);
+        if (less(_lower_bound, row.position())) {
```
Comment (haaawk) [inline cache_flat_mutation_reader.hh:624]: When can this be false?

**Review by haaawk — APPROVED** [review]
LGTM, I just have one question about when _lower_bound can be not smaller than row.position().

File: cache_flat_mutation_reader.hh (hunk 1 above)
Comment (tgrabiec) [reply cache_flat_mutation_reader.hh:624]: For example, when the query range starts with a position which is equal to dummy entry's position.

**Pull request merged**